packages/angular_devkit/build_facade/src/workspace-entities/configuration.ts

```diff
@@ -0,0 +1,32 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import { BaseException, JsonObject } from '@angular-devkit/core';
+
+
+export class MissingConfigurationBuildFacadeException extends BaseException {
+  constructor(props: JsonObject) {
+    super(`Configuration does not contain or inherit build facade: ${props}.`);
+  }
+}
+
+export class Configuration {
+  constructor(readonly name: string, readonly props: JsonObject) {
+    if (!this.props.buildFacade) {
```

Comment:
Again, I think these (Workspace, Project, Target and Configuration) should be interfaces that are validated by the Architect instance against their respective Schema, but there's no reason to have `x.props.xyz` to access `xyz`.  Look at how I did it in Schematic for making it extensible (the definition types).

Reply:
Yeah I agree their implementation isn't very good now. This was the first bit I made and it ended up not very good. Fixing.